### DISCUSSION TITLE
ESP32C3U UART0 not working

### DISCUSSION THREAD

**Discussion by giancarlo64** [micropython/micropython#10505] [ESP32]
Hi, I was trying to run a communication using the UART0 on a M5stamp C3U (ESP32C3U) with no success.  I have initialized the UART as follow:

uart = UART(0, baudrate=115200, bits=8, stop=1)

but if i send some chars  like

uart.write('hello\n') 

I see nothing coming out of the TX pin (using a DSO)

I have also tried to move the UART0  to other pins but I still get no data out 

uart = UART(0, baudrate=115200, tx=3, rx=4, bits=8, stop=1)

I have to change the UART from 0 to 1 and after it run:

uart = UART(1, baudrate=115200, tx=3, rx=4, bits=8, stop=1)

Why?

I have also another problem. If inside the program I want to output some data on the shell of Thonny IDE the data will not be visualized. I have to do a Control+C and after that the data will be visualized and the program continue to run.

I have the following firmware:

esp32c3-usb-20220618-v1.19.1

Thanks for your help

Gian Carlo

**Comment by Leo5126** [comment]
Hi @giancarlo64,

Regarding your first question about the uart interface. Here is my guess.
Several interfaces are stored in the firmware, which you can access with the so-called ID. This ID is the first parameter during initialization that you have changed from 0 to 1. I guess that the pins you use are for the interface with the ID 1. 
On this page you can find more settings for UART: https://docs.micropython.org/en/latest/library/machine.UART.html?highlight=uart#methods

To your second question. Which command are you using?

Greetings, 

Leo

**Comment by ShanmuHuang** [comment]
I have the same problem.

**Comment by kjm1102** [comment]
Are you sure about the DSO check, 6 characters at 115Kbaud is going to be super quick. What level does the Tx pin idle at at when you're not using it? What happens if you over ride the UART(0) config with a Tx=Pin(1, Pin.OUT) followed by Tx(1) then Tx(0)?
